ch2/exercise2-3/popcount.go

```diff
@@ -0,0 +1,30 @@
+package popcount
+
+// pc[i] is the population count of i.
+var pc [256]byte
+
+func init() {
+	for i := range pc {
+		pc[i] = pc[i/2] + byte(i&1)
+	}
+}
+
+// PopCount returns the population count (number of set bits) of x.
+func PopCount(x uint64) int {
+	return int(pc[byte(x>>(0*8))] +
+		pc[byte(x>>(1*8))] +
+		pc[byte(x>>(2*8))] +
+		pc[byte(x>>(3*8))] +
+		pc[byte(x>>(4*8))] +
+		pc[byte(x>>(5*8))] +
+		pc[byte(x>>(6*8))] +
+		pc[byte(x>>(7*8))])
+}
+
+func PopCountLoop(x uint64) int {
+	var count byte
+	for i := 8; i > 0; i-- {
+		count = count + pc[byte(x>>((i-1)*8))]
+	}
```

Comment:
Out of curiosity: Why do you go 8 -> 1 (really 7 -> 0) instead of 0 -> 7.  This would simplify the `i -1` and have your loop ascending which is more common.

Reply:
I'm not really sure why I did that tbh